home/models.py

```diff
@@ -295,7 +295,29 @@ def clean(self, value):
                 f"{self.MEDIA_CAPTION_MAX_LENGTH} characters, your message is "
                 f"{len(result['message'])} characters long"
             )
+
+        # find variables
+        vars_in_msg = re.findall(r"{{(.*?)}}", result["message"])
+        non_digit_variables = [var for var in vars_in_msg if not var.isdigit()]
+        if non_digit_variables != []:
+            errors["message"] = ValidationError("Please enter numeric variables only.")
+        # check variable order
+        digit_variables = [var for var in vars_in_msg if var.isdigit()]
+        if digit_variables != []:
+            if (
+                all(
+                    x < y
+                    for x, y in zip(digit_variables, digit_variables[1:], strict=False)
+                )
+                is not True
+                or digit_variables[0] == "0"
```

Comment:
Instead of having this all in one giant if statement, maybe it's better to separate it out. At the moment it's rather difficult to figure out what this code is doing at a glance.

A cleaner way of doing this might be to create a list of what we expect the values of the variables to be, and then to compare that to the list that we extracted from the message. If they don't equal, then we know we need to raise a validation error. eg:
```
expected_variables = [str(i + 1) for i in range(len(vars_in_msg))]
if vars_in_message != expected_variables:
    errors....
```